frontends/web/src/routes/account/summary/accountssummary.tsx

```diff
@@ -65,9 +68,61 @@ export const AccountsSummary = ({
   const [accountsTotalBalance, setAccountsTotalBalance] = useState<accountApi.TAccountsTotalBalance>();
   const [coinsTotalBalance, setCoinsTotalBalance] = useState<accountApi.TCoinsTotalBalance>();
   const [balances, setBalances] = useState<Balances>();
+  const { lightningConfig } = useLightning();
 
   const hasCard = useSDCard(devices);
 
+
+  const [lightningBalance, setLightningBalance] = useState<accountApi.IBalance>();
+  const fetchLightningBalance = useCallback(async () => {
+    setLightningBalance(await getLightningBalance());
+  }, []);
+
+  // if there is an active lightning account derived from this keystore, subscribe to any node state changes.
+  useEffect(() => {
+    const lightningAccounts = lightningConfig.accounts;
+    if (lightningAccounts.length) {
+      fetchLightningBalance();
+      const subscriptions = [subscribeNodeState(fetchLightningBalance)];
+      return () => unsubscribe(subscriptions);
+    }
+  }, [fetchLightningBalance, lightningConfig]);
+
+  const lightningKeystoreName = useLoad(() => {
+    const account = lightningConfig.accounts[0];
+    return account ? accountApi.keystoreName(account.rootFingerprint) : Promise.resolve(null);
+  }, [lightningConfig]);
+
+
+  // lightning account exists but is not from any connected or remembered keystores
+  const hasLightningFromOtherKeystore = (
+    lightningConfig.accounts.length !== 0
+    && (
+      !accountsByKeystore.some(({ keystore }) => {
+        return keystore.rootFingerprint === lightningConfig.accounts[0].rootFingerprint;
+      })
+    )
+  );
+  let keystores = accountsByKeystore.length;
+  let lightningAccountKeystoreName = '';
+  let keystoreDisambiguatorName;
+  if (hasLightningFromOtherKeystore) {
```

Comment:
This whole `if` condition is not very readable, IMO. Can you try to simplify a bit? Maybe it could be useful to return an empty string instead of `null` in the `lightningKeystoreName` useLoad?